### PR TITLE
Dna enhancement

### PR DESCRIPTION
**Pull request by meg-1**
### Describe your change:

adding @appgurueu 's reccomendations from #6323 

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: strings/dna.py [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
-    'Invalid Strand'
+    >>> dna_matching_strand("GFGG")
+    Traceback (most recent call last):
+    Exception: Invalid Strand
     """
 
-    r = len(re.findall("[ATCG]", dna)) != len(dna)
-    val = dna.translate(dna.maketrans("ATCG", "TAGC"))
-    return "Invalid Strand" if r else val
+    is_strand = len(re.findall("[ATCG]", dna)) != len(dna)
```
Comment (CaedenPH) [inline strings/dna.py:21]: Do you mean `is_not_strand`? Or perhaps `invalid_strand`

File: strings/dna.py [hunk 2]
```diff
@@ -1,25 +1,29 @@
...
+    strand = dna.translate(dna.maketrans("ATCG", "TAGC"))
```
Comment (CaedenPH) [inline strings/dna.py:22]: You can move this after the if statement and remove the else

File: strings/dna.py (hunk 1 above)
Comment (meg-1) [reply strings/dna.py]: thank you for your feedback! changed the variable name according to your recommendation!

File: strings/dna.py (hunk 2 above)
Comment (meg-1) [reply strings/dna.py:22]: hi! thank you for pointing that out as well! i think the code you saw might be one of the older versions of this file. in the newer versions i use an "raise exception" approach.

**Review by CaedenPH — APPROVED** [review]
@cclauss Looks good

**Review by cclauss — APPROVED** [review]

**Pull request closed without merging**